### ISSUE TITLE
Screen reader is not announcing selected/unselected state information for the menu items inside the 'Pin column' dropdown menu in Windows.

### ISSUE THREAD

**Issue by msftedad** [ceolter/ag-grid#10632]
### Link to reproducible scenario

https://www.ag-grid.com/react-data-grid/pivoting/

### Describe the bug

Steps to reproduce 

1. Open the URL: [React Grid: Pivoting | AG Grid](https://www.ag-grid.com/react-data-grid/pivoting/) in the latest edge browser.
2. Pivoting page will appear.
3. Press tab key to reach to the 'more options' control in the column header of the table and press enter. Context menu will appear.
4. Press arrow keys to reach to the 'Pin column' control and activate it using enter key. drop-down will appear.
5. Press arrow keys to reach to the menu items.
6. Verify whether Screen reader is announcing selected/unselected state information for the menu items inside the 'Pin column' dropdown menu or not.

Actual behaviour
Using arrow keys, screen reader is announcing it as 'No pin 1 of 3' when focus moves to the menu items inside the 'Pin column' dropdown menu in Windows.

https://github.com/user-attachments/assets/50a966b9-0ba3-40ea-a0e4-c7e816c23bc8

![Image](https://github.com/user-attachments/assets/828bee06-7b1c-4729-b82f-054432748d68)

Expected behaviour
Using arrow keys, screen reader should announce it as 'No pin 1 of 3 selected' when focus moves to the menu items inside the 'Pin column' dropdown menu in Windows.

More information

### Version

33.3.0

### Does the issue occur for a specific framework only?

All languages

### Is the issue only observable on a specific browser?

All browsers

**Comment by ava-utting** [comment]
Hi @msftedad ,

Thank you for bringing this to our attention.

Currently there's no way to implement this behavior with AG Grid. However, we agree this would be a valuable addition to the product and we have added a feature request for this.

We have added this requirement to our backlog and we are tracking it with the following reference:
AG-15457 

We have no timeline for this at the moment, but this can still be picked up by our team when they're making modifications to this functional area of AG Grid. Please use the functionality currently provided by AG Grid until this is delivered at a later point.

The best way to track this is to sign up for AG Grid new release notifications using [the instructions here](https://ag-grid.zendesk.com/hc/en-us/articles/360002648058-How-do-I-sign-up-for-ag-Grid-release-notifications-). This way you'll know as soon as a new version is out and you can check whether this specific item was implemented on [the changelog page](https://www.ag-grid.com/ag-grid-changelog/).

Thanks again for bringing this up with us.

Kind regards,
Ava
